chemist-aws/src/main/scala/AwsDiscovery.scala

```diff
@@ -314,16 +325,19 @@ class AwsDiscovery(
 
     val datacenter = in.getPlacement.getAvailabilityZone
 
-    val mirrorTemplate: Option[String] =
-      machineTags.get(AwsTagKeys.mirrorTemplate) orElse defaultInstanceTemplate
+    //Note: if node is not labeled explicitly we ignore it to avoid wasting time trying to continiously
+    // reconnect to incompatible instances
+    val mirrorTemplate: Option[String] = machineTags.get(AwsTagKeys.mirrorTemplate)
 
-    for {
+    val r = for {
       a <- toLocation(dns, datacenter, mirrorTemplate, Mirroring)
-      // _  = log.debug(s"discovered mirrioring template '$a'")
+      // _  = log.debug(s"discovered mirroring template '$a'")
     } yield AwsInstance(
       id = in.getInstanceId,
       tags = machineTags,
       locations = NonEmptyList(a)
     )
+
+    r.leftMap(message => (in.getInstanceId, message))
```

Comment:
@stew @timperrett 

Do you mean pass in `in.getInstanceId` and `machineTags` and rename `toLocation` to something like `toAwsInstance`? Does it really improve readability? What is the purpose of `fromAWSInstance` then? extract machine tags? In this case it seem more sense to inline for comprehension from toLocation ...

how about one of those versions:

````
    toLocation(dns, datacenter, mirrorTemplate, Mirroring).map {
      a => AwsInstance(
        id = in.getInstanceId,
        tags = machineTags,
        locations = NonEmptyList(a)
      )
    }.leftMap(message => (in.getInstanceId, message))

    //my favorite
    toLocation(dns, datacenter, mirrorTemplate, Mirroring).map(
      a => AwsInstance(in.getInstanceId, machineTags, locations = NonEmptyList(a))
    ).leftMap(message => (in.getInstanceId, message))

    //this one is a bit ugly
    toLocation(dns, datacenter, mirrorTemplate, Mirroring) match {
      case \/-(a) => \/-(AwsInstance(
        id = in.getInstanceId,
        tags = machineTags,
        locations = NonEmptyList(a)
      ))
      case -\/(message) => -\/((in.getInstanceId, message))
    }
````